pkgs/cupertino_http/lib/src/cupertino_client.dart

```diff
@@ -366,6 +371,9 @@ class CupertinoClient extends BaseClient {
     request.headers.forEach(urlRequest.setValueForHttpHeaderField);
     final task = urlSession.dataTaskWithRequest(urlRequest);
     final subscription = StreamController<Uint8List>(onCancel: () {
+      final taskTracker = _tasks[task];
+      if (taskTracker == null) return;
```

Comment:
This will skip `task.cancel()` if there's no tracker. Is that intended? If not, I think you can simplify this to `_tasks[task]?.responseListenerCancelled = true;`

Reply:
The tracker is removed when the request is completed, in which case cancelling the response stream is a no-op (because no more data will arrive).